packages/amplication-server/src/core/entity/entity.service.ts

```diff
@@ -222,43 +234,80 @@ export class EntityService {
     return newEntity;
   }
 
-  async createInitialEntities(appId: string, user: User): Promise<Entity[]> {
-    const app = this.prisma.app.update({
+  async createDefaultEntities(appId: string, user: User): Promise<void> {
+    return this.bulkCreateEntities(appId, user, DEFAULT_ENTITIES);
+  }
+
+  /**
+   * Creates multiple entities.
+   * The function supports creation of entities with reference to other entities.
+   * Use "[entityName]" notation for any property values and it will be replaced with the actual ID of the entity, whether an existing one or a new one.
+   * Reference to new entities must only be located after the declaration of the new entity (the order of the entities count)
+   */
+  async bulkCreateEntities(
+    appId: string,
+    user: User,
+    entities: BulkEntityData[]
+  ): Promise<void> {
+    const existingEntities = await this.prisma.entity.findMany({
       where: {
-        id: appId
-      },
-      data: {
-        entities: {
-          create: INITIAL_ENTITIES.map(entity => ({
-            name: entity.name,
-            displayName: entity.displayName,
-            pluralDisplayName: entity.pluralDisplayName,
-            description: entity.description,
-            lockedAt: new Date(),
-            lockedByUser: {
-              connect: {
-                id: user.id
-              }
-            },
-            entityVersions: {
-              create: {
-                commit: undefined,
-                versionNumber: CURRENT_VERSION_NUMBER,
-                name: entity.name,
-                displayName: entity.displayName,
-                pluralDisplayName: entity.pluralDisplayName,
-                description: entity.description,
-                entityFields: {
-                  create: entity.fields
-                }
+        appId: appId
+      }
+    });
+
+    //create a map of existing entities
+    const entityMap = Object.fromEntries(
+      existingEntities.map(entity => {
+        return [`[${entity.name}]`, entity.id];
+      })
+    );
+
+    //replace any reference to entity (using the "[entityName]" notation) with the actual ID of the entity
+    for (const entity of entities) {
+      JSON.parse(JSON.stringify(entity), (key, value) => {
+        if (entityMap[value]) {
+          return entityMap[value];
+        } else {
+          return value;
+        }
+      });
```

Comment:
I don't understand this code.

Reply:
From the context of its use I understand this complexity can be eliminated with pre-defined IDs.

Reply:
Pre-defined IDs are just "ok" for this scenario. This implementation allows us to use the "bulkCreate" in any future need without knowing the ID of the entity.

Reply:
This code stringify the object,  and runs a replace for any placeholder that exists in the map object,
After all the values were replaced it re-parses the JSON string to an object.